src/components/search/search.jsx

```diff
@@ -0,0 +1,172 @@
+import { cn } from '@/utilities/functions';
+import { File, Folder, Search, SquareSlash } from 'lucide-react';
+import { useState, useEffect } from 'react';
+import { Label } from '..';
+import Loader from '../loader';
+
+const SearchBox = ( props ) => {
+	const {
+		onChange = () => { },
+		handleResultClick = () => { },
+		placeholder = 'Search',
+		variant = 'primary', // primary, secondary, ghost
+		disabled = false,
+		size = 'sm', // sm, md, lg
+		loading = false,
+		loadingIcon = <Loader />, // Default loading icon
+		searchResult = [],
+		additionalResult = [],
+		searchResultIcon = <File />,
+		additionalResultLabel = 'Category',
+		additionalResultIcon = <Folder />,
+		className,
+		children,
+	} = props;
+
+	const [ isOpen, setIsOpen ] = useState( false );
+
+	// Close dropdown on click outside
+	useEffect( () => {
+		const handleClickOutside = ( event ) => {
+			if ( ! event.target.closest( '.searchbox-wrapper' ) ) {
+				setIsOpen( false );
+			}
+		};
+		document.addEventListener( 'mousedown', handleClickOutside );
+		return () => {
+			document.removeEventListener( 'mousedown', handleClickOutside );
+		};
+	}, [] );
+
+	const commonClass = 'relative w-full focus-within:z-10';
+	const baseClasses = 'font-normal placeholder-text-tertiary text-text-secondary w-full focus:outline-none';
+
+	const sizeClasses = {
+		sm: 'p-1 rounded-md text-xs [&>svg]:h-4 [&>svg]:w-4',
+		md: 'p-1.5 rounded-md text-sm [&>svg]:h-5 [&>svg]:w-5',
+		lg: 'p-2 rounded-lg text-base [&>svg]:h-6 [&>svg]:w-6',
```

Comment:
Instead of specifying both width `w` and height `h` for the SVG, you can use `[&>svg]:size` to apply the same dimensions uniformly.